extensions/transport_sockets/tls/ssl_socket.cc

```diff
@@ -247,6 +307,16 @@ void SslSocket::shutdownSsl() {
   }
```

Comment:
`shutdownSsl()` might be called by someone else and assert on `handshake_complete_` when interrupting an asynchronous handshake.